KakapoExample/KakapoTests/KakapoDBTests.swift

```diff
@@ -277,6 +437,79 @@ class KakapoDBTests: QuickSpec {
+            }
+            
+            it("should not deadlock when synchronously deleting the database from another queue during a write operation") {
+                let user = sut.insert { (id) -> UserFactory in
+                    dispatch_sync(queue) {
+                        sut.delete(sut.find(UserFactory.self, id: 0)!)
+                    }
+                    return UserFactory(id: id, db: sut)
+                }
+                
+                let users = sut.findAll(UserFactory.self)
+                
+                expect(user).toEventuallyNot(beNil())
+                expect(users.count).toEventually(equal(1))
+            }
+            
+            it("should not deadlock when deleting into database during a reading operation") {
+                let result = sut.filter(UserFactory.self, includeElement: { (_) -> Bool in
+                    sut.delete(sut.find(UserFactory.self, id: 0)!)
+                    return true
+                })
+                
+                let users = sut.findAll(UserFactory.self)
+                
+                expect(result).toEventuallyNot(beNil())
+                expect(users.count).toEventually(equal(0))
+            }
+            
+            it("should not deadlock when updating into database during a reading operation") {
+                let result = sut.filter(UserFactory.self, includeElement: { (_) -> Bool in
+                    sut.update(UserFactory(firstName: "Alex", lastName: "Manzella", age: 30, id: 0))
+                    return true
+                })
+                
+                let user = sut.findAll(UserFactory.self).first!
+                
+                expect(result).toEventuallyNot(beNil())
+                expect(user.firstName).toEventually(equal("Alex"))
+                expect(user.lastName).toEventually(equal("Manzella"))
+                expect(user.age).toEventually(equal(30))
+            }
+            
+            it("should be able to concurrently delete objects") {
+                let users = sut.create(UserFactory.self, number: 100)
+
+                dispatch_apply(100, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) { i in
+                    print(i)
+                    sut.delete(users[i])
+                }
+
+                // We still have 1 user created in beforeEach
+                expect(sut.findAll(UserFactory.self).count).to(equal(1))
+            }
+            
+            it("should be able to concurrently update objects") {
+                let users = sut.create(UserFactory.self, number: 100)
+                
+                dispatch_apply(100, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0)) { i in
+                    sut.update(users[i])
+                }
+                
+                // Updated 100 users + the one created in beforeEach
+                expect(sut.findAll(UserFactory.self).count).to(equal(101))
```

Comment:
I think this one is wrong, wouldn't fail but update the wrong stuff , to make user I was also deleting